plugin/core/collections.py

```diff
@@ -131,6 +132,16 @@ def update(self, d: Dict[str, Any]) -> None:
             else:
                 self.set(key, value)
 
+    def resolve(self, variables: Dict[str, str]) -> "DottedDict":
+        """
+        Resolve a DottedDict that may potentially contain template variables like $folder
```

Comment:
Even though the `returns` documentation mentions it, it could be clearer from the name that it creates a copy instead of updating itself. Maybe `create_resolved()`?

It would potentially be even clearer if this would be a class method to which you would be a dict.

Reply:
I went with create_resolved. I am too lazy to type out the name DottedDict if it were a classmethod :)